push.sh

```diff
@@ -15,7 +15,11 @@ fi
 # If username and password are provided, add auth
 if [ ! -z "$PLUGIN_USERNAME" ] && [ ! -z "$PLUGIN_PASSWORD" ]; then
 
-    AUTH="--user '${PLUGIN_USERNAME}':'${PLUGIN_PASSWORD}'"
+    #escape
+    PLUGIN_USERNAME=$(echo ${PLUGIN_USERNAME} | tr -d ';`()')
+    PLUGIN_PASSWORD=$(echo ${PLUGIN_PASSWORD} | tr -d ';`()')
```

Comment:
Could you explain this a bit?

I think spaces in passwords is unlikely to be an issue, but the quotes are intended to avoid the command from parsing `--user user:pass word` as `--user user:pass` with an additional argument `word`.

It shouldn't matter `'` or `"` but simply:

    AUTH="--user \"${PLUGIN_USERNAME}:${PLUGIN_PASSWORD}\""

should do. The main difference being that the entire arg is wrapped with quotes rather than each variable. This will still achieve the goal of preventing splitting.